torchvision/csrc/io/image/cpu/decode_png.cpp

```diff
@@ -52,14 +54,20 @@ torch::Tensor decode_png(
 
   struct Reader {
     png_const_bytep ptr;
+    png_size_t count;
   } reader;
   reader.ptr = png_const_bytep(datap) + 8;
+  reader.count = datap_len - 8;
 
   auto read_callback =
       [](png_structp png_ptr, png_bytep output, png_size_t bytes) {
         auto reader = static_cast<Reader*>(png_get_io_ptr(png_ptr));
+        if (reader->count < bytes) {
+          png_error(png_ptr, "Out of bound read in decode_png");
```

Comment:
So instead of showing this error, PyTorch throws an "Internal Error" message without any more details.

I wonder if here it would be advised to do something like:
```suggestion
          TORCH_CHECK(false, "Out of bound read in decode_png");
```